feat: add self-monitor support for Log Agent (#2144)
kyma-project/telemetry-manager: docs/user/logs.md

**Symptom**:

- No logs arrive at the backend.
- In the LogPipeline status, the `TelemetryFlowHealthy` condition has status **GatewayAllTelemetryDataDropped** or **AgentAllTelemetryDataDropped**

**Cause**: Incorrect backend endpoint configuration (such as using the wrong authentication credentials) or the backend is unreachable.

**Solution**:

> 1. Check the `telemetry-log-gateway`  Pod (in case of **GatewayAllTelemetryDataDropped** status) or `telemetry-log-agent` Pod (in case of **AgentAllTelemetryDataDropped** status) for error logs by calling `kubectl logs -n kyma-system {POD_NAME}`.

Reviewer: ```suggestion
1. Check the error logs for the affected Pod by calling `kubectl logs -n kyma-system {POD_NAME}`:
   - For **GatewayAllTelemetryDataDropped**, check Pod `telemetry-log-gateway`
   - For **AgentAllTelemetryDataDropped**, check Pod `telemetry-log-agent`
```

**Symptom**:

- The backend is reachable and the connection is properly configured, but some logs are refused.
- In the LogPipeline status, the `TelemetryFlowHealthy` condition has status **GatewaySomeTelemetryDataDropped** or **AgentSomeTelemetryDataDropped**.

**Cause**: It can happen due to a variety of reasons - for example, the backend is limiting the ingestion rate.

**Solution**:

1. Check the `telemetry-log-gateway` Pod (in case of **GatewaySomeTelemetryDataDropped**) or `telemetry-log-agent` Pod (in case of **AgentSomeTelemetryDataDropped**) for error logs by calling `kubectl logs -n kyma-system {POD_NAME}`. Also, check your observability backend to investigate potential causes.

> 2. If backend is limiting the rate by refusing logs, try the options desribed in [Buffer Filling Up](#gateway-buffer-filling-up).

Reviewer: ```suggestion
3. If the backend is limiting the rate by refusing logs, try the options described in [Buffer Filling Up](#gateway-buffer-filling-up).
```

> 3. Otherwise, take the actions appropriate to the cause indicated in the logs.

Reviewer: ```suggestion
4. Otherwise, take the actions appropriate to the cause indicated in the logs.
```

Author: Why change it from 3 to 4?
It is indeed the 3rd item in the steps
